nodeletctl/pkg/nodeletctl/deployer.go

```diff
@@ -286,15 +286,24 @@ func (nd *NodeletDeployer) CreatePf9User() error {
 		zap.S().Infof("User name already exist")
 		return nil
 	}
-	var cmdList []string
 	zap.S().Infof("Creating pf9 user")
-	cmdList = append(cmdList, "mkdir -p /opt/pf9/home")
-	cmdList = append(cmdList, "groupadd -f pf9group")
-	cmdList = append(cmdList, "id -u pf9 &>/dev/null || useradd -d /opt/pf9/home -G pf9group pf9")
 
-	for _, cmd := range cmdList {
-		if _, _, err := nd.client.RunCommand(cmd); err != nil {
-			return fmt.Errorf("createPf9User: %s: %s", cmd, err)
+	_, _, err = nd.client.RunCommand("mkdir -p /opt/pf9/home")
+	if err != nil {
+		return fmt.Errorf("failed to create home dir: %s", err)
+	}
+
+	_, _, err = nd.client.RunCommand("groupadd -f pf9group")
+	if err != nil {
+		return fmt.Errorf("failed to add pf9group: %s", err)
+	}
+
+	_, _, err = nd.client.RunCommand("id -u pf9")
+	if err != nil {
+		zap.S().Infof("check for id pf9: %s", err)
+		_, _, err = nd.client.RunCommand("useradd -d /opt/pf9/home -G pf9group pf9")
+		if err != nil {
+			return fmt.Errorf("createPf9User: %s", err)
```

Comment:
Earlier we had a chance to log the cmd getting executed also, in case of failure. 
Now we aren't. Better to log for debugging purpose?

Reply:
I tried to print the errors according to the commands and also the RunCommand further prints the command which is being run [here](https://github.com/platform9/nodelet/blob/480223486f52c1407683d38448b780e7dfe9da76/nodeletctl/pkg/nodeletctl/localExecutor.go#L26). So, let me know if we want to update this.

Reply:
Ok, thanks @vedantjoshi84. Should be fine then.
lgtm